contentcuration/contentcuration/models.py

```diff
@@ -120,6 +120,11 @@ def generate_file_on_disk_name(checksum, filename):
         os.makedirs(directory)
     return os.path.join(directory, h + ext.lower())
 
+def generate_storage_url(filename):
+    """ Returns place where file is stored """
+    h, ext = os.path.splitext(filename)
+    return "{}/{}/{}/{}".format(settings.STORAGE_URL[1:-1], h[0], h[1], h + ext.lower())
```

Comment:
This should return an absolute URL, since in theory we may want to set STORAGE_URL to something like `https://learningequality.s3.amazonaws.com/`, if we hosted files elsewhere. We may want to normalize for it having or not having a trailing slash, though, so I would recommend:

- Replace `STORAGE_URL[1:-1]` with `STORAGE_URL.rstrip("/")`.
- You may need to modify client side code to handle the absolute URL.